qdev_wrappers/customised_instruments/AWG5014_ext.py

```diff
@@ -4,12 +4,39 @@
 
 
 class AWG5014_ext(Tektronix_AWG5014):
-    def __init__(self, name, visa_address, **kwargs):
-        super().__init__(name, visa_address, **kwargs)
+    def __init__(self, name, address, **kwargs):
+        super().__init__(name, address, **kwargs)
         self.add_parameter(name='current_seq',
                            parameter_class=ManualParameter,
                            initial_value=None,
                            label='Uploaded sequence index',
                            vals=vals.Ints())
-        self.ref_source('EXT')
+        self.add_parameter(name='seq_mode',
+                           set_cmd=self._set_seq_mode,
+                           get_cmd=self._get_seq_mode)
         self.clear_message_queue()
+
+    def _set_seq_mode(self, status):
+        if str(status).upper() in ['TRUE', '1', 'ON']:
+            for i in range(self.sequence_length()):
+                self.set_sqel_loopcnt(i, 1)
+        elif str(status).upper() in ['FALSE', '0', 'OFF']:
+            for i in range(self.sequence_length()):
+                self.set_sqel_loopcnt_to_inf(i)
+
+    def _get_seq_mode(self):
+        if self.get_sqel_loopcnt() == '1':
+            return False
+        else:
+            return True
+
+    def send_and_load_awg_file(self, awg_file, filename):
```

Comment:
This function looks general enough that it perhaps belongs in the driver?

Reply:
Possibly. I think up to taste to be honest, 'make' is already in there as are 'send' and 'load' and a few combinations. Maybe @WilliamHPNielsen has an opinion?